clevertap-core/src/main/java/com/clevertap/android/sdk/Utils.java

```diff
@@ -179,6 +189,94 @@ public static Bitmap getBitmapFromURL(@NonNull String srcUrl) {
         }
     }
 
+    public static Bitmap getBitmapFromURLWithSizeConstraint(String srcUrl, int size) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.